blocks/query/edit.tsx

```diff
@@ -1,17 +1,362 @@
 import { __ } from '@wordpress/i18n';
-import { useBlockProps } from '@wordpress/block-editor';
+import { useEffect, useState } from '@wordpress/element';
+
+import { InnerBlocks, InspectorControls, useBlockProps } from '@wordpress/block-editor';
+import {
+  PanelBody,
+  PanelRow,
+  RangeControl,
+  SelectControl,
+  TextControl,
+} from '@wordpress/components';
+import ApiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+import { PostPicker, TermSelector, useDebounce } from '@alleyinteractive/block-editor-tools';
+import type { WP_REST_API_Post, WP_REST_API_Posts } from 'wp-types';
 
 import './index.scss';
 
+interface EditProps {
+  attributes: {
+    numberOfPosts?: number;
+    minNumberOfPosts?: number;
+    maxNumberOfPosts?: number;
+    offset?: number;
+    postTypes?: string[];
+    posts?: any[];
+    terms?: {
+      [key: string]: any[];
+    };
+    searchTerm?: string;
+    query: {
+      [key: string]: string | number | number[] | string[];
+    }
+  };
```

Comment:
These attributes don't seem to match what's defined in block.json (min/max number of posts is missing, and block.json also has `pages`).